build changes - go dep, doc updates, purge fixed, namespace plugin moved

josmo opened the pull request:
@ipedrazas I can separate out all the changes in different prs however here's the summary

* using drone to build/publish (allows simple forks to do personal builds so they can use new features till it gets merged)
* Moved over to go dep till 1.11 figures out the next official thing
* split the docs to a diff file
* gofmt
* moved the public into a separate namespace
* fixed the purge value being ignored on delete
* docs update for purge and helm_command


generally if you accept the pr and use drone to build/publish you only need a couple secrets (docker_username, docker_password, plugin_repo) and you're good to go :)

ipedrazas commented:
This is massive 😂😂😂

I’ve just realised I’ve,lost visibility on the tests... which is not great. I’m currently on holidays so, I have very limited time. Anything you could do to minimise the time I need to spend checking this PR  would be highly appreciated.

josmo commented:
@ipedrazas  I can definitely create a few smaller PRs (let me know if that's easier), but I'll summarize all the changes which if I split would probably represent the separate PRs

1. go dep - This is most of the file changes. So vendor folder is now ignored and deleted. Does require the ```dep ensure``` prior to tests and builds

2. Changed to make the plugin a lib from a prior PR. Tests got moved into the plugin module so running them with ```go test ./... -v -cover``` will still run all the same tests

3. the .drone.yml file has the deps, test, build, and publishing in it. It's generic so that any fork can easily build and publish their own version of the plugin but just enabling the fork in their drone and setting up a couple secrets. It also has the update to tag and build versions in the binary. also using the cli you can just ```DRONE_REPO_OWNER=ipedrazas DRONE_REPO_NAME=drone-helm drone exec``` to build the product locally

4. Split out the usage into docs to make the readme a bit simpler to view

5. Updated kubectl to 1.11.0

6. Fixed purge value not being initialized into the plugin.

7. gofmt

If you're down for one big PR. There's two updates left. Changing the couple places in the readme that reference my repo, and the build.sh needs to place the build product where the dockerfile expects it since I was using drone to build it.
